docs/index.html

```diff
@@ -288,20 +288,54 @@ <h2>#TwitterThrowback</h2>
     var submitButton = document.getElementById('submit-button');
     var loadingIndicator = document.getElementById('loading-indicator');
 </script>
+<script>
+    /**
+     * @function getMonthAndDay
+     * @description returns the Date API Friendly month and day 
+     * @param monthAndDayString a string containing the month and day separated by a space
+    */
+     function getMonthAndDay(monthAndDayString) {
+        const todayAsArray = [new Date().getMonth() + 1, new Date().getDate()];
+        if (monthAndDayString.split(' ').length < 2) {
+            monthAndDayString = monthAndDayString.concat(` ${todayAsArray[1]}`);
+        }
+        const dummyDate = new Date(monthAndDayString);
+        return [ dummyDate.getMonth() + 1, dummyDate.getDate()];
+    }
+</script>
 <script>
     // Prefill the value if the user is coming from a direct link (oldtweets.today/#username)
     // Normally, I'd want to do this using a path, but GitHub Pages doesn't support routing via paths
+    
+    
     let hash = window.location.hash.split('#')[1];
     if (hash != '' && hash != undefined) {
-        usernameInput.value = hash;
+        const startingPointOfDate = hash.indexOf('?');
+        if (startingPointOfDate > -1) {
+            usernameInput.value = hash.substring(0, startingPointOfDate);
+            const monthAndDay = hash.substring(startingPointOfDate + 1).replace('date=', '').trim().replace('+', ' ');
+            month.value = getMonthAndDay(monthAndDay)[0].toString();
+            day.value = getMonthAndDay(monthAndDay)[1];
+        } else {
+            usernameInput.value = hash;
+        }
     } else {
         // Also support query string (oldtweets.today?username=<username>)
         let queryString = window.location.search;
         let queryParams = new URLSearchParams(queryString);
         if (queryParams.get('username')) {
-            usernameInput.value = queryParams.get('username');
+            // since ?username=<username>?date=<date>
```

Comment:
Why would this be? Why not `&date=<date>`. This is the canonical query string format, and I don't think we should support otherwise. `date` is an "initialization", just like username. Neither should have a special status

Reply:
you can enter the app via either `#username=username` or `?username=<username>`. 
`&username=<username>?date=<date>` is actually the correct way. But i think because of the way twitter sharing treats urls, they have `&hashtags` as a param, specifying `oldtweets.today&username=<username>` was giving issues. I think thats why the `?username=<username>` was used. Now though, line #473 fixes those issues with
```   const encodedUri = encodeURIComponent(`?username=${window.username}?date=${month.options[month.selectedIndex].textContent}+${day.value}`); ```

Reply:
Hold up...I think you're missing something. Probably you made a few typos.

- `&username=<username>?date=<date>` is not correct. Did you mean `?username=<username>&date=<date>` ? Because that should be the proper way to access this.
- Twitter link sharing shouldn't cause any problems. If you need to share a url with a `&`, '#' or `?`, you're supposed to urlencode them (and it looks like that's what you're doing with `encodeURIComponent`).


My question is: why are you writing `?username=${window.username}?date=${month.options[month.selectedIndex].textContent}+${day.value}` instead of `?username=${window.username}&date=${month.options[month.selectedIndex].textContent}+${day.value}` ?

Reply:
I mean, a query string starts with `?~ for the first parameter and `&` for subsequent ones. Why are you using two sets of query strings (`?` and `?`) in one url instead of joining the parameters with `?` and `&`? 😕

Reply:
Yh, that was a typo. I was carried away by the hash's representation of the params which is currently  `#username?date=<date>`. Would update PR.

Reply:
Oh, okay. Good.